sdks/java/io/file-schema-transform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformFormatProviders.java

```diff
@@ -61,6 +73,49 @@ static MapElements<Row, GenericRecord> mapRowsToGenericRecords(Schema beamSchema
         .via(AvroUtils.getRowToGenericRecordFunction(AvroUtils.toAvroSchema(beamSchema)));
   }
 
+  // Applies generic mapping from Beam row to other data types through the provided mapFn.
+  // Implemenets error handling with metrics and DLQ support.
+  // Arguments:
+  //    name: the metric name to use.
+  //    mapFn: the mapping function for mapping from Beam row to other data types.
+  //    outputTag: TupleTag for output. Used to direct output to correct output source, or in the
+  //        case of error, a DLQ.
+  static class ErrorCounterFn<OutputT extends Object> extends DoFn<Row, OutputT> {
```

Comment:
this name doesn't seem right. This transform is a .map() that aggregates errors on a separate tag

Reply:
What would be a better name in this case? BeamRowMapperWithDlq something like this?

Reply:
Done. Tests have been added for the class.

Reply:
error handling beam row mapper maybe? that or your suggestion works

Reply:
also, ideally we would have a unit test for this isolated from any particular format. I do appreciate that you've added tests that use this for all the formats.

Reply:
Done. Changed name to `BeamRowMapperWithDlq`, and added a generic test independent of any format.